tools/Customization/DevHome.Customization/Views/FileExplorerPage.xaml.cs

```diff
@@ -1,23 +1,83 @@
-// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT License.
-
-using DevHome.Common.Extensions;
-using DevHome.Customization.ViewModels;
-using Microsoft.UI.Xaml;
-using Microsoft.UI.Xaml.Controls;
-
-namespace DevHome.Customization.Views;
-
-public sealed partial class FileExplorerPage : Page
-{
-    public FileExplorerViewModel ViewModel
-    {
-        get;
-    }
-
-    public FileExplorerPage()
-    {
-        ViewModel = Application.Current.GetService<FileExplorerViewModel>();
-        this.InitializeComponent();
-    }
-}
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using DevHome.Common.Extensions;
+using DevHome.Common.Services;
+using DevHome.Customization.ViewModels;
+using DevHome.FileExplorerSourceControlIntegration.Services;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace DevHome.Customization.Views;
+
+public sealed partial class FileExplorerPage : Page
+{
+    public FileExplorerViewModel ViewModel
+    {
+        get;
+    }
+
+    private string? RootPath
+    {
+        get; set;
+    }
+
+    private static readonly System.Buffers.SearchValues<char> InvalidChars = System.Buffers.SearchValues.Create("<>*?|.");
+
+    public FileExplorerPage()
+    {
+        ViewModel = Application.Current.GetService<FileExplorerViewModel>();
+        this.InitializeComponent();
+        var experimentationService = Application.Current.GetService<IExperimentationService>();
+        var isEnabled = experimentationService.IsFeatureEnabled("FileExplorerSourceControlIntegration");
+        if (isEnabled)
+        {
+            TrackRepository.Visibility = Visibility.Visible;
+            DisplayTrackRepository.Visibility = Visibility.Visible;
+        }
+    }
+
+    public void AddButton_Click(object sender, RoutedEventArgs e)
+    {
+        RootPath = RootPathTextBox.Text;
+
+        var isValidated = ValidateRepositoryPath(RootPath);
+        if (isValidated)
+        {
+            // TO DO: Determine if the extension GUID should be stored instead of name as it is more identifiable and/or determine any other unique property to use for

```

Comment:
Does this TODO have an issue?

Reply:
> Super nit-y, but I think there's VS goodness if you actually use the word "TODO" instead of "TO DO"

Dittor for clang-format.

Reply:
Yes, this TODO item is being tracked as part of work for an ADO task. Thanks!